src/test/shared/ui/prompter.test.ts

```diff
@@ -37,13 +37,15 @@ describe('Prompter', function () {
         await assert.rejects(prompter.prompt())
     })
 
-    it('can attach multiple callbacks', async function () {
+    it('can attach multiple transformations', async function () {
         const prompter = new SimplePrompter(1)
-        prompter.after(resp => resp * 2)
-        prompter.after(resp => resp + 2)
-        prompter.after(resp => resp * 2)
-        prompter.after(resp => resp - 2)
-        prompter.transform(resp => resp.toString()).after(resp => `result: ${resp}`)
+        prompter
+            .transform(resp => resp * 2)
+            .transform(resp => resp + 2)
+            .transform(resp => resp * 2)
+            .transform(resp => resp - 2)
+            .transform(resp => resp.toString())
+            .transform(resp => `result: ${resp}`)
```

Comment:
I guess this is what you were hinting at when you mentioned wanting something like pipe()? Since this is so abstract we might be able to use a library (later/todo).

Reply:
Yeah. Though in for `Prompter` these transformations are deferred until `prompt` is called. I believe `Prompter` could be considered a kind of generic monad in this case, where `prompt` unboxes the underlying and `transform` is the binding function.